torch/_inductor/constant_folding.py

```diff
@@ -195,3 +203,60 @@ def constant_fold(gm, constraint_fn: Optional[Callable[[torch.fx.Node], bool]] =
     gm.graph.eliminate_dead_code()
     gm.graph.lint()
     gm.recompile()
+
+
+@torch.utils._python_dispatch._disable_current_modes()
+def constant_graph_tag(gm: torch.fx.GraphModule):
+    cf = ConstantFolder(gm, skip_constructors=True)
+    cf.run()
+
+    for node in gm.graph.nodes:
+        if (
+            node.op == "get_attr"
+            or node in cf.node_replacements
+            or node in cf.replaced_uses
+        ):
+            node.tag = CONST_MODULE_TAG
+        else:
+            node.tag = MODULE_TAG
+
+
+def get_constant_graph(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    """
+    Construct a GraphModule which corresponds to the part which could be
+    constant folded in provided gm.
+    """
+
+    constant_graph_tag(gm)
+    # We rewrite the tags, if it's a constant being directly consumed, without
+    # any folding opportunity, we keep it in main gm.
+    for node in gm.graph.nodes:
+        if node.op == "get_attr":
+            used_to_fold = False
+            for u in node.users:
+                if u.tag == CONST_MODULE_TAG:
+                    used_to_fold = True
+                    break
+            if not used_to_fold:
+                node.tag = MODULE_TAG
```

Comment:
Can't we remove the `node.op == "get_attr"` condition above in `constant_graph_tag` and remove this loop?

Reply:
I don't think so, the "users" will only be tagged when all inputs are being tagged. So if this is happening at tagging time, right at this moment, the users aren't determined whether it's in const or main yet. Or are you asking to move this loop to `constant_graph_tag`?